### PR TITLE
Rules: Command to move a replication rule; Closes #133

### LINKED ISSUE #133: Option to "move" a rule in CLI
Motivation
----------
The idea is basically the same principle as with BB8. A specific rule should be moved to another RSE, including all it's properties.

_Migrated from [JIRA](https://its.cern.ch/jira/browse/RUCIO-2721)_

### PR DESCRIPTION
**Pull request by bari12**
Rules: Command to move a replication rule; Closes #133

### CODE REVIEW

**Comment by vingar** [comment]
please squash commits

File: bin/rucio [hunk 1]
```diff
@@ -2091,6 +2091,18 @@ def update_rule(args):
...
 
+@exception_handler
+def move_rule(args):
+    """
+    %(prog)s move-rule [options] <ruleid> <rse_expression>
+
+    Update a rule.
+    """
+    client = get_client(args)
+    print client.move_replication_rule(rule_id=args.rule_id, rse_expression=args.rse_expression)
```
Comment (vingar) [inline bin/rucio:2102]: print vs. logging ?

Comment (bari12) [reply bin/rucio:2102]: For now we always print the output on the command line; If we want to do that through logging, I would say we should consistently change this everywhere in the CLI?

File: lib/rucio/client/ruleclient.py [hunk 2]
```diff
@@ -143,6 +143,25 @@ def reduce_replication_rule(self, rule_id, copies, exclude_expression=None):
...
+        :raises:                    RuleNotFound, RuleReplaceFailed
+        """
+
+        path = self.RULE_BASEURL + '/' + rule_id + '/move'
+        url = build_url(choice(self.list_hosts), path=path)
+        data = dumps({'rule_id': rule_id, 'rse_expression': rse_expression})
+        r = self._send_request(url, type='POST', data=data)
+        if r.status_code == codes.created:
+            return loads(r.text)
+        else:
```
Comment (vingar) [inline lib/rucio/client/ruleclient.py]: useless else:

File: lib/rucio/core/rule.py [hunk 3]
```diff
@@ -1317,6 +1317,68 @@ def reduce_rule(rule_id, copies, exclude_expression=None, session=None):
...
+        if rule.state != RuleState.OK:
+            raise RuleReplaceFailed('The source rule must be in state OK.')
+
+        if rule.grouping == RuleGrouping.ALL:
+            grouping = 'ALL'
+        elif rule.grouping == RuleGrouping.NONE:
+            grouping = 'NONE'
+        else:
+            grouping = 'DATASET'
+
```
Comment (vingar) [inline lib/rucio/core/rule.py]: Some pythonic way of doing this: 
grouping =  {RuleGrouping.ALL: 'ALL', RuleGrouping.NONE: 'NONE'}.get(rule.grouping, 'DATASET')

less readable though

**Comment by vingar** [comment]
conflict. rebase

**Comment by bari12** [comment]
This cannot be rebased, i will have to merge it manually.

File: lib/rucio/client/ruleclient.py (hunk 2 above)
Comment (bari12) [reply lib/rucio/client/ruleclient.py]: True, but I went for consistency here, as it is like this in all clients.

File: lib/rucio/core/rule.py (hunk 3 above)
Comment (bari12) [reply lib/rucio/core/rule.py]: I like it 😄

**Review by vingar — DISMISSED** [review]
one minor comment about if statement

File: lib/rucio/client/ruleclient.py (hunk 2 above)
Comment (vingar) [reply lib/rucio/client/ruleclient.py]: yes and I always try to remove it at few places. Not blocking anyway

Comment (bari12) [reply lib/rucio/client/ruleclient.py]: True. I'll change it.

**Comment by bari12** [comment]
I will resubmit these two PR. They are somehow completely stuck in travis.

**Pull request closed without merging**